packages/contracts-bedrock/test/L1/OPPrestateUpdater.t.sol

```diff
@@ -0,0 +1,352 @@
+        inputs[0] = OPContractsManager.OpChainConfig(
+            chainDeployOutput.systemConfigProxy, chainDeployOutput.opChainProxyAdmin, Claim.wrap(bytes32(0))
+        );
+
+        address proxyAdminOwner = chainDeployOutput.opChainProxyAdmin.owner();
+        vm.etch(address(proxyAdminOwner), vm.getDeployedCode("test/mocks/Callers.sol:DelegateCaller"));
+
+        vm.expectRevert(OPPrestateUpdater.PDGPrestateRequired.selector);
+        DelegateCaller(proxyAdminOwner).dcForward(
+            address(prestateUpdater), abi.encodeCall(OPPrestateUpdater.updatePrestate, (inputs))
+        );
+    }
+
+    function test_deploy_notImplemented_reverts() public {
+        OPContractsManager.DeployInput memory input = OPContractsManager.DeployInput({
+            roles: OPContractsManager.Roles({
+                opChainProxyAdminOwner: address(0),
+                systemConfigOwner: address(0),
+                batcher: address(0),
+                unsafeBlockSigner: address(0),
+                proposer: address(0),
+                challenger: address(0)
+            }),
+            basefeeScalar: 0,
+            blobBasefeeScalar: 0,
+            l2ChainId: 0,
+            startingAnchorRoot: bytes(abi.encode(0)),
+            saltMixer: "",
+            gasLimit: 0,
+            disputeGameType: GameType.wrap(0),
+            disputeAbsolutePrestate: Claim.wrap(0),
+            disputeMaxGameDepth: 0,
+            disputeSplitDepth: 0,
+            disputeClockExtension: Duration.wrap(0),
+            disputeMaxClockDuration: Duration.wrap(0)
+        });
+
+        vm.expectRevert(OPPrestateUpdater.NotImplemented.selector);
+        prestateUpdater.deploy(input);
+    }
+
+    function test_upgrade_notImplemented_reverts() public {
+        opChainConfigs.push(
+            OPContractsManager.OpChainConfig({
+                systemConfigProxy: ISystemConfig(address(0)),
+                proxyAdmin: IProxyAdmin(address(0)),
+                absolutePrestate: Claim.wrap(0)
+            })
+        );
+
+        vm.expectRevert(OPPrestateUpdater.NotImplemented.selector);
+        prestateUpdater.upgrade(opChainConfigs);
+    }
+
+    function test_addGameType_notImplemented_reverts() public {
+        gameInput.push(
+            OPContractsManager.AddGameInput({
+                saltMixer: "hello",
+                systemConfig: ISystemConfig(address(0)),
+                proxyAdmin: IProxyAdmin(address(0)),
+                delayedWETH: IDelayedWETH(payable(address(0))),
+                disputeGameType: GameType.wrap(2000),
+                disputeAbsolutePrestate: Claim.wrap(bytes32(hex"deadbeef1234")),
+                disputeMaxGameDepth: 73,
+                disputeSplitDepth: 30,
+                disputeClockExtension: Duration.wrap(10800),
+                disputeMaxClockDuration: Duration.wrap(302400),
+                initialBond: 1 ether,
+                vm: IBigStepper(address(opcm.implementations().mipsImpl)),
```

Comment:
```suggestion
                vm: IBigStepper(address(0)),
```

Reply:
If I sent address 0 it reverts with DeploymentFailed

Reply:
that's weird... what in here would do that?

https://github.com/ethereum-optimism/optimism/blob/c6c6711329e36d1c4c70f1bc3b770551d5ce058e/packages/contracts-bedrock/src/L1/OPPrestateUpdater.sol#L77-L80